GitHubReactions.ps1

```diff
@@ -0,0 +1,493 @@
+        GitHub.Reaction
+
+    .OUTPUTS
+        GitHub.Reaction
+
+    .EXAMPLE
+        Set-GitHubReaction -OwnerName PowerShell -RepositoryName PowerShell -IssueNumber 12626 -ReactionType rocket
+
+        Sets the 'rocket' reaction for issue 12626 of the PowerShell\PowerShell project.
+
+    .EXAMPLE
+        Get-GitHubPullRequest -Uri https://github.com/microsoft/PowerShellForGitHub -PullRequest 193 | Set-GitHubReaction -ReactionType Heart
+
+        Gets a GitHub pull request and pipes it into Set-GitHubReaction to set the 'heart' reaction for that pull request.
+
+    .NOTES
+        Currently, this only supports reacting to issues and pull requests. Issue comments, commit comments and PR comments will come later.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "", Justification="One or more parameters (like NoStatus) are only referenced by helper methods which get access to it from the stack via Get-Variable -Scope 1.")]
+    param(
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName='Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName='Elements', ValueFromPipelineByPropertyName)]
+        [Parameter(Mandatory, ParameterSetName='Uri', ValueFromPipelineByPropertyName)]
+        [Alias('Issue')]
+        [Alias('PullRequest')]
+        [Alias('PullRequestNumber')]
+        [int64] $IssueNumber,
+
+        [ValidateSet('+1', '-1', 'Laugh', 'Confused', 'Heart', 'Hooray', 'Rocket', 'Eyes')]
+        [Parameter(Mandatory)]
+        [string] $ReactionType,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/issues/$IssueNumber/reactions"
+    $description = "Setting reaction $ReactionType for Issue $IssueNumber in $RepositoryName"
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Description' =  $description
+        'Method' = 'Post'
+        'Body' = @{ content = $ReactionType.ToLower() } | ConvertTo-Json
```

Comment:
Please _don't_ pipe input `ConvertTo-Json`.  Only call it with `-InputObject` parameter.  This is a very intentional call pattern throughout the module due to the unintended consequences of piping objects into `ConvertTo-Json`.  Specifically, it will "unwrap" single object arrays:

```powershell
@(1) | ConvertTo-Json
1

ConvertTo-Json -InputObject @(1)
[
   1
]
```

That might not have any material impact on _this_ call, but it can (and has) on other calls.  Given that new functions often start out as copy/pastes of existing functions, we want to avoid unnecessary head-scratching errors introduced in those new functions by bad patterns introduced in the existing functions.